x-pack/plugins/security_solution/public/management/pages/trusted_apps/service/index.ts

```diff
@@ -5,186 +5,4 @@
-      body: JSON.stringify(request),
-    });
-  }
-
-  async updateTrustedApp(
-    params: PutTrustedAppsRequestParams,
-    updatedTrustedApp: PutTrustedAppUpdateRequest
-  ) {
-    return this.http.put<PutTrustedAppUpdateResponse>(
-      resolvePathVariables(TRUSTED_APPS_UPDATE_API, params),
-      { body: JSON.stringify(updatedTrustedApp) }
-    );
-  }
-
-  async getTrustedAppsSummary(request: GetTrustedAppsSummaryRequest) {
-    return this.http.get<GetTrustedAppsSummaryResponse>(TRUSTED_APPS_SUMMARY_API, {
-      query: request,
-    });
-  }
-
-  getPolicyList(options?: Parameters<typeof sendGetEndpointSpecificPackagePolicies>[1]) {
-    return sendGetEndpointSpecificPackagePolicies(this.http, options);
-  }
-
-  /**
-   * Assign a policy to trusted apps. Note that Trusted Apps MUST NOT be global
-   *
-   * @param policyId
-   * @param trustedApps[]
-   */
-  assignPolicyToTrustedApps(
-    policyId: string,
-    trustedApps: MaybeImmutable<TrustedApp[]>
-  ): Promise<PutTrustedAppUpdateResponse[]> {
-    return this._handleAssignOrRemovePolicyId('assign', policyId, trustedApps);
-  }
-
-  /**
-   * Remove a policy from trusted apps. Note that Trusted Apps MUST NOT be global
-   *
-   * @param policyId
-   * @param trustedApps[]
-   */
-  removePolicyFromTrustedApps(
-    policyId: string,
-    trustedApps: MaybeImmutable<TrustedApp[]>
-  ): Promise<PutTrustedAppUpdateResponse[]> {
-    return this._handleAssignOrRemovePolicyId('remove', policyId, trustedApps);
-  }
-
-  private _handleAssignOrRemovePolicyId(
-    action: 'assign' | 'remove',
-    policyId: string,
-    trustedApps: MaybeImmutable<TrustedApp[]>
-  ): Promise<PutTrustedAppUpdateResponse[]> {
-    if (policyId.trim() === '') {
-      throw new Error('policy ID is required');
-    }
-
-    if (trustedApps.length === 0) {
-      throw new Error('at least one trusted app is required');
-    }
-
-    return pMap(
-      trustedApps,
-      async (trustedApp) => {
-        if (isGlobalEffectScope(trustedApp.effectScope)) {
-          throw new Error(
-            `Unable to update trusted app [${trustedApp.id}] policy assignment. It's effectScope is 'global'`
-          );
-        }
-
-        const policies: string[] = !isGlobalEffectScope(trustedApp.effectScope)
-          ? [...trustedApp.effectScope.policies]
-          : [];
-
-        const indexOfPolicyId = policies.indexOf(policyId);
-
-        if (action === 'assign' && indexOfPolicyId === -1) {
-          policies.push(policyId);
-        } else if (action === 'remove' && indexOfPolicyId !== -1) {
-          policies.splice(indexOfPolicyId, 1);
-        }
-
-        return this.updateTrustedApp(
-          { id: trustedApp.id },
-          {
-            ...toUpdateTrustedApp(trustedApp),
-            effectScope: {
-              type: 'policy',
-              policies,
-            },
-          }
-        );
-      },
-      P_MAP_OPTIONS
-    );
-  }
-}
+export * from './trusted_apps_http_service';
```

Comment:
What if we remove this file all together and import from this new file instead of using barrel re-exports?

Recently we discovered all those jest memory leaks that were coming because of this pattern.

Reply:
Let's hold off on that for now. I would just like to focus on getting this in for 8.0 so we can remove the API and rather not incur additional refactors

Reply:
sounds good! would you mind opening an issue or adding a comment to not forget about it? (I will forget about it, I am sure)